### PR TITLE
chore(ffi): Log panics from spawned tokio tasks

### PR DESCRIPTION
**Pull request by jplatte**
This should help us track down recent mutex poisioning issues.

### CODE REVIEW

File: bindings/matrix-sdk-ffi/src/helpers.rs [hunk 1]
```diff
@@ -1,5 +1,33 @@
...
+
+pub fn spawn_future<F>(future: F) -> JoinHandle<()>
+where
+    F: Future<Output = ()> + Send + 'static,
+{
+    // tokio::spawn doesn't require unwind safety either, everything we do seems
+    // to be unwind-unsafe but if one of these tasks panics we usually have
+    // bigger issues than inconsistent state (note: that is the worst that can
+    // happen by asserting unwind safety, no unsafe code is involved here)
+    let future = AssertUnwindSafe(future);
```
Comment (Hywan) [inline bindings/matrix-sdk-ffi/src/helpers.rs:21]: It reminds me something O:-).

**Review by Hywan — APPROVED** [review]

**Comment by jplatte** [comment]
Actually we are already using a crate called `log-panics` on Android and it seems like after fixing / working around log subscription issues on Android, that crate took care of printing the panic. So I think this PR is actually pointless.

**Comment by poljar** [comment]
> Actually we are already using a crate called `log-panics` on Android and it seems like after fixing / working around log subscription issues on Android, that crate took care of printing the panic. So I think this PR is actually pointless.

But we don't use `log-panics` on iOS, wouldn't it be useful there and at the same time remove the need for `log-panics`?

**Comment by jplatte** [comment]
Sure, it might be useful on iOS as well but I think I'd prefer the much more general approach taken by `log-panics`. There's no reason we can't use it on both, is there? Or, if we want to hook into tracing without going through, we can check if there's an existing alternative crate, or swap out the log creation bits from `log-panics` (it's a tiny crate).

**Comment by poljar** [comment]
Yeah hooking it into `tracing` sounds good, there's https://github.com/matrix-org/matrix-rust-sdk/pull/1502 which attempts to streamline the log setups between the various platforms.

**Pull request closed without merging**